Strapi import with Postgres seems to ignore custom schema

## Bug report

### Required System information

- Node.js version: 16.20.0
- NPM version: 8.19.4
- Strapi version: 4.10.5
- Database: Postgres
- Operating system: Linux
- Is your project Javascript or Typescript: Javascript

### Describe the bug

I'm trying to import data that was exported from a different strapi instance. Both strapi instances use the same strapi configuration and the same database configuration. I'm using the same custom database schema for both, defined in database.js.

When importing data I get the following error:

`error: [ERROR] insert into "admin_permissions_role_links" ("permission_id", "permission_order", "role_id") values ($1, $2, $3) - relation "admin_permissions_role_links" does not exist`

But the "admin_permissions_role_links" table exists in the custom schema. 

When changing the configuration of the target instance to not use a custom schema (so it will automatically use the public schema) and rebuild strapi, the import works successfully.

### Expected behavior

Importing should also work with custom postgres schemas

### Screenshots

![error screenshot](https://github.com/strapi/strapi/assets/7661210/fa7aca9f-55ba-406f-b71f-e0830de1948c)

--- a/packages/core/data-transfer/src/strapi/queries/link.ts
+++ b/packages/core/data-transfer/src/strapi/queries/link.ts
@@ -198,7 +198,7 @@ export const createLinkQuery = (strapi: Strapi.Strapi, trx?: Knex.Transaction) =
       if (attribute.joinColumn) {
         const joinColumnName = attribute.joinColumn.name;
 
-        // Note: this addSchema may not be necessary, but it can't hurt
+        // Note: this addSchema may not be necessary, but is added for safety
         const qb = connection(addSchema(metadata.tableName))
           .where('id', left.ref)
           .update({ [joinColumnName]: right.ref });

--- a/packages/core/data-transfer/src/strapi/queries/link.ts
+++ b/packages/core/data-transfer/src/strapi/queries/link.ts
@@ -197,7 +197,9 @@ export const createLinkQuery = (strapi: Strapi.Strapi, trx?: Knex.Transaction) =
 
       if (attribute.joinColumn) {
         const joinColumnName = attribute.joinColumn.name;
-        const qb = connection(metadata.tableName)
+
+        // Note: this addSchema may not be necessary, but it can't hurt
+        const qb = connection(addSchema(metadata.tableName))
           .where('id', left.ref)
           .update({ [joinColumnName]: right.ref });
         if (trx) {

--- a/packages/core/data-transfer/src/strapi/queries/link.ts
+++ b/packages/core/data-transfer/src/strapi/queries/link.ts
@@ -8,6 +8,12 @@ export const createLinkQuery = (strapi: Strapi.Strapi, trx?: Knex.Transaction) =
   const query = () => {
     const { connection } = strapi.db;
 
+    // TODO: Export utils from database and use the addSchema that is already written
+    const addSchema = (tableName: string) => {
+      const schemaName = connection.client.connectionSettings.schema;
+      return schemaName ? `${schemaName}.${tableName}` : tableName;
+    };
+
     async function* generateAllForAttribute(uid: string, fieldName: string): AsyncGenerator<ILink> {
       const metadata = strapi.db.metadata.get(uid);
 
@@ -31,7 +37,10 @@ export const createLinkQuery = (strapi: Strapi.Strapi, trx?: Knex.Transaction) =
       if (attribute.joinColumn) {
         const joinColumnName: string = attribute.joinColumn.name;
 
-        const qb = connection.queryBuilder().select('id', joinColumnName).from(metadata.tableName);
+        const qb = connection
+          .queryBuilder()
+          .select('id', joinColumnName)
+          .from(addSchema(metadata.tableName));
 
         if (trx) {
           qb.transacting(trx);
@@ -65,7 +74,7 @@ export const createLinkQuery = (strapi: Strapi.Strapi, trx?: Knex.Transaction) =
           inverseOrderColumnName,
         } = attribute.joinTable;
 
-        const qb = connection.queryBuilder().from(name);
+        const qb = connection.queryBuilder().from(addSchema(name));
 
         type Columns = {
           left: { ref: string | null; order?: string };
@@ -253,7 +262,7 @@ export const createLinkQuery = (strapi: Strapi.Strapi, trx?: Knex.Transaction) =
 
         assignOrderColumns();
 
-        const qb = connection.insert(payload).into(name);
+        const qb = connection.insert(payload).into(addSchema(name));
         if (trx) {
           qb.transacting(trx);
         }

